container-engine-lib/lib/backend_interface/objects/service_directory/persistent_directories.go

```diff
@@ -1,13 +1,19 @@
 package service_directory
 
 type DirectoryPersistentKey string
+type DirectoryPersistentSize int64
+
+type PersistentDirectory struct {
+	PersistentKey DirectoryPersistentKey
+	Size          DirectoryPersistentSize
+}
 
 type PersistentDirectories struct {
-	ServiceDirpathToDirectoryPersistentKey map[string]DirectoryPersistentKey
+	ServiceDirpathToPersistentDirectory map[string]PersistentDirectory
```

Comment:
same here

Reply:
I think this has always been like this to allow for serialization into DB; we should figure out a more idiomatic way for immutable structs that need to be serialized; otherwise we need to write marshall/unmarshall